File: docs/versioned_docs/version-1.11.0/API-Reference/python-examples/api-monitor/delete-messages.py
import os
from uuid import UUID

import requests

base = os.environ.get("LANGFLOW_URL", "")
api_key = os.environ.get("LANGFLOW_API_KEY", "")
flow_id = os.environ.get("FLOW_ID", "")

headers = {"accept": "*/*", "Content-Type": "application/json", "x-api-key": api_key}

list_resp = requests.get(
    f"{base}/api/v1/monitor/messages",
    headers=headers,
    params={"flow_id": flow_id},
    timeout=30,
)
list_resp.raise_for_status()
messages = list_resp.json()
if not messages:
    print("No messages to delete.")
    raise SystemExit(0)

ids = [UUID(str(m["id"])) for m in messages[:2]]
params = [("message_ids", str(i)) for i in ids]

response = requests.delete(f"{base}/api/v1/monitor/messages", headers=headers, params=params, timeout=30)
response.raise_for_status()
print(response.status_code)

File: docs/versioned_docs/version-1.11.0/API-Reference/javascript-examples/api-monitor/delete-messages.js
const url = `${process.env.LANGFLOW_URL ?? ""}/api/v1/monitor/messages`;

const options = {
  method: 'DELETE',
  headers: {
    "accept": `*/*`,
    "Content-Type": `application/json`,
    "x-api-key": `${process.env.LANGFLOW_API_KEY ?? ""}`,
  },
  body: JSON.stringify([
  "MESSAGE_ID_1",
  "MESSAGE_ID_2"
]),
};

fetch(url, options)
  .then(async (response) => {
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    const text = await response.text();
    console.log(text);
  })
  .catch((error) => console.error(error));
